client/client.go

```diff
@@ -208,23 +191,9 @@ func (c *Client) update(latestRoot bool) (data.TargetFiles, error) {
 	}
 	snapshotMetas, err := c.decodeSnapshot(snapshotJSON)
 	if err != nil {
-		// ErrRoleThreshold could indicate snapshot keys have been
-		// revoked, so retry with the latest root.json
-		if isDecodeFailedWithErrRoleThreshold(err) && !latestRoot {
-			return c.updateWithLatestRoot(nil)
-		}
 		return nil, err
 	}
 
-	// If we don't have the root.json, download it, save it in local
-	// storage and restart the update
```

Comment:
Removing this and enforcing the new root rotation on each update is a breaking change with the [implemented workflow spec 0.9.9 ](https://github.com/theupdateframework/tuf/blob/v0.9.9/docs/tuf-spec.txt#L714-L737).

I think this should be explicitly mentioned in the PR description in case some still rely on it.